src/gui/src/clockWidget.cpp

```diff
@@ -809,6 +817,58 @@ void ClockTreeView::fit()
   fitInView(scene_->sceneRect(), Qt::KeepAspectRatio);
 }
 
+ClockNodeGraphicsViewItem* ClockTreeView::getItemFromName(
+    const std::string& name)
+{
+  if (items_.find(name) == items_.end()) {
+    return nullptr;
+  }
+
+  return items_[name];
+}
+
+std::set<ClockNodeGraphicsViewItem*> ClockTreeView::getNodes(
+    const SelectionSet& selections)
+{
+  std::set<ClockNodeGraphicsViewItem*> nodes;
+  for (const auto& selection : selections) {
+    ClockNodeGraphicsViewItem* item = getItemFromName(selection.getName());
+    if (item != nullptr) {
+      nodes.insert(item);
+    }
+  }
+
+  return nodes;
+}
+
+int ClockTreeView::findInCts(const SelectionSet& selections)
+{
+  std::set<ClockNodeGraphicsViewItem*> nodes = getNodes(selections);
+  if (!nodes.empty()) {
+    // remove old selection
+    clearSelection();
+    for (auto node : nodes) {
+      node->setSelected(true);
+    }
+    fitSelection();
+  }
+  return (int) nodes.size();
```

Comment:
This function is doing a lot of things: changing the current selection, fitting the new selection and returning the number of selected nodes. The name findInCts doesn't say much about what is being done.

One alternative could be to use a name like changeAndFitSelection() and you could keep the number of selected nodes in a member variable such as `number_of_selected_nodes_` to be used elsewhere.

Reply:
Fixed.
I changed the method `ClockTreeView::findInCts` to `ClockTreeView::changeSelection`.
Now the method returns a boolean indicating if the selection changed.

The method `ClockWidget::findInCts` calls `ClockTreeView::fitSelection` when the selection changes.

Since the number of currently selected nodes isn't used yet, I didn't implement a member variable, but it can be easily implemented as a method:
`int ClockTreeView::numberOfSelectedNodes() { (int) scene_->selectedItems().size(); };`